### ISSUE TITLE
Allow the use of Linked Data URIs for Islandora taxonomies instead of term IDs

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#148]
Currently, Workbench configuration files require the use of term IDs for `media_use_tid` and `model`. Problem with term IDs is that people creating config files need to look them up. If Workbench could understand term URIs, they could be used instead. The benefit to using URIs over term IDs is that the URIs are standardized across all Islandora instances.

The View described in https://github.com/Islandora/documentation/issues/1311 could be enabled by Islandora Workbench Integration, and Workbench could query it to get the term ID it needs internally to assign the taxonomy term.

**Comment by mjordan** [comment]
Views FTW! This View:

```
langcode: en
status: true
dependencies:
  config:
    - user.role.administrator
  module:
    - rest
    - serialization
    - taxonomy
    - user
id: term_id_from_uri
label: 'Term from URI'
module: views
description: ''
tag: ''
base_table: taxonomy_term_field_data
base_field: tid
display:
  default:
    display_plugin: default
    id: default
    display_title: Master
    position: 0
    display_options:
      access:
        type: role
        options:
          role:
            administrator: administrator
      cache:
        type: tag
        options: {  }
      query:
        type: views_query
        options:
          disable_sql_rewrite: false
          distinct: false
          replica: false
          query_comment: ''
          query_tags: {  }
      exposed_form:
        type: basic
        options:
          submit_button: Apply
          reset_button: false
          reset_button_label: Reset
          exposed_sorts_label: 'Sort by'
          expose_sort_order: true
          sort_asc_label: Asc
          sort_desc_label: Desc
      pager:
        type: mini
        options:
          items_per_page: 10
          offset: 0
          id: 0
          total_pages: null
          expose:
            items_per_page: false
            items_per_page_label: 'Items per page'
            items_per_page_options: '5, 10, 25, 50'
            items_per_page_options_all: false
            items_per_page_options_all_label: '- All -'
            offset: false
            offset_label: Offset
          tags:
            previous: ‹‹
            next: ››
      style:
        type: serializer
      row:
        type: fields
        options:
          inline: {  }
          separator: ''
          hide_empty: false
          default_field_elements: true
      fields:
        name:
          id: name
          table: taxonomy_term_field_data
          field: name
          entity_type: taxonomy_term
          entity_field: name
          label: ''
          alter:
            alter_text: false
            make_link: false
            absolute: false
            trim: false
            word_boundary: false
            ellipsis: false
            strip_tags: false
            html: false
          hide_empty: false
          empty_zero: false
          type: string
          settings:
            link_to_entity: true
          plugin_id: term_name
          relationship: none
          group_type: group
          admin_label: ''
          exclude: false
          element_type: ''
          element_class: ''
          element_label_type: ''
          element_label_class: ''
          element_label_colon: true
          element_wrapper_type: ''
          element_wrapper_class: ''
          element_default_classes: true
          empty: ''
          hide_alter_empty: true
          click_sort_column: value
          group_column: value
          group_columns: {  }
          group_rows: true
          delta_limit: 0
          delta_offset: 0
          delta_reversed: false
          delta_first_last: false
          multi_type: separator
          separator: ', '
          field_api_classes: false
          convert_spaces: false
      filters:
        status:
          value: '1'
          table: taxonomy_term_field_data
          field: status
          plugin_id: boolean
          entity_type: taxonomy_term
          entity_field: status
          id: status
          expose:
            operator: ''
            operator_limit_selection: false
            operator_list: {  }
          group: 1
        field_external_uri_uri:
          id: field_external_uri_uri
          table: taxonomy_term__field_external_uri
          field: field_external_uri_uri
          relationship: none
          group_type: group
          admin_label: ''
          operator: 'not empty'
          value: ''
          group: 1
          exposed: false
          expose:
            operator_id: field_external_uri_uri_op
            label: 'External URI (field_external_uri:uri)'
            description: ''
            use_operator: false
            operator: field_external_uri_uri_op
            operator_limit_selection: false
            operator_list: {  }
            identifier: field_external_uri_uri
            required: false
            remember: false
            multiple: false
            remember_roles:
              authenticated: authenticated
              anonymous: '0'
              administrator: '0'
              fedoraadmin: '0'
            placeholder: ''
          is_grouped: false
          group_info:
            label: ''
            description: ''
            identifier: ''
            optional: true
            widget: select
            multiple: false
            remember: false
            default_group: All
            default_group_multiple: {  }
            group_items: {  }
          plugin_id: string
      sorts: {  }
      header: {  }
      footer: {  }
      empty: {  }
      relationships: {  }
      arguments:
        field_external_uri_uri:
          id: field_external_uri_uri
          table: taxonomy_term__field_external_uri
          field: field_external_uri_uri
          relationship: none
          group_type: group
          admin_label: ''
          default_action: default
          exception:
            value: all
            title_enable: false
            title: All
          title_enable: false
          title: ''
          default_argument_type: query_parameter
          default_argument_options:
            query_param: uri
            fallback: ''
            multiple: and
          default_argument_skip_url: false
          summary_options:
            base_path: ''
            count: true
            items_per_page: 25
            override: false
          summary:
            sort_order: asc
            number_of_records: 0
            format: default_summary
          specify_validation: false
          validate:
            type: none
            fail: 'not found'
          validate_options: {  }
          glossary: false
          limit: 0
          case: none
          path_case: none
          transform_dash: false
          break_phrase: false
          plugin_id: string
      display_extenders: {  }
    cache_metadata:
      max-age: -1
      contexts:
        - 'languages:language_content'
        - 'languages:language_interface'
        - request_format
        - url
        - url.query_args
        - user.roles
      tags: {  }
  rest_export_1:
    display_plugin: rest_export
    id: rest_export_1
    display_title: 'REST export'
    position: 1
    display_options:
      display_extenders: {  }
      path: term_from_uri
      pager:
        type: none
        options:
          offset: 0
      style:
        type: serializer
        options:
          uses_fields: false
          formats:
            json: json
      row:
        type: data_entity
        options: {  }
      auth:
        - basic_auth
    cache_metadata:
      max-age: -1
      contexts:
        - 'languages:language_content'
        - 'languages:language_interface'
        - request_format
        - url
        - user.roles
      tags: {  }
```

will allow queries like `curl -v -uadmin:islandora "http://localhost:8000/term_from_uri?_format=json&uri=http://purl.org/coar/resource_type/c_c513"`, from which we can get the term ID.

**Comment by mjordan** [comment]
This feature will require disabling the Islandora Workbench Integration module, then pulling in updates from its Github repo, then re-enabling it.

**Comment by mjordan** [comment]
Resolved with 33ef6b666beea00484417822920a565fa7280b9c. Documentation is in the new "Using term URIs instead of term IDs" section of the readme.
